models/transaction.go

```diff
@@ -1,20 +1,125 @@
 package models
 
 import (
-	"bytes"
 	"encoding/hex"
 	"fmt"
-	"github.com/onflow/flow-go-sdk"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	gethTypes "github.com/ethereum/go-ethereum/core/types"
-	"github.com/ethereum/go-ethereum/rlp"
 	"github.com/onflow/cadence"
-	cdcCommon "github.com/onflow/cadence/runtime/common"
 	"github.com/onflow/flow-go/fvm/evm/types"
 )
 
+type Transaction interface {
+	Hash() (common.Hash, error)
```

Comment:
add a todo here, we don't need error as return value, the reason it is here is because direct call return an error but it doesn't have to (it's always nil) https://github.com/onflow/flow-go/blob/master/fvm/evm/types/call.go#L79

Reply:
Good catch :clap: I didn't notice that it was always returning `nil`. Added in https://github.com/onflow/flow-evm-gateway/pull/132/commits/5a48c66bd8ff4d562cb836256d0cb44044937737